solr/core/src/test/org/apache/solr/util/TestSolrJacksonAnnotation.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.util;
+
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.solr.SolrTestCase;
+import org.apache.solr.common.annotation.Property;
+import org.apache.solr.common.util.JsonSchemaCreator;
+import org.apache.solr.common.util.JsonSchemaValidator;
+import org.apache.solr.common.util.Utils;
+
+public class TestSolrJacksonAnnotation extends SolrTestCase {
+
+  public void testSerDe() throws Exception {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.setAnnotationIntrospector(new SolrPropertyAnnotationInspector());
+
+    TestObj o = new TestObj();
+    o.field = "v1";
+    o.f2 = "v2";
+    o.ifld = 1234;
+    String json = mapper.writeValueAsString(o);
+
+    Map m = (Map) Utils.fromJSONString(json);
+    assertEquals("v1",  m.get("field"));
+    assertEquals("v2",  m.get("friendlyName"));
+    assertEquals("1234",  String.valueOf(m.get("friendlyIntFld")));
+    TestObj o1 = mapper.readValue(json, TestObj.class);
+
+    assertEquals("v1", o1.field);
+    assertEquals("v2", o1.f2);
+    assertEquals(1234, o1.ifld);
+
+    Map<String, Object> schema = JsonSchemaCreator.getSchema(TestObj.class);
```

Comment:
Overall, this is fine but I thought I'd share that my preference for testing objects is simply a toString and equals comparison against a string literal.  This is way easier to write and understand (IMO), less code, and is easy to see as more _complete_ than by-field assertEquals since you don't need to ask yourself if mysteriously other stuff is in there that is not tested.  But I understand this is a matter of taste and you are perfectly welcome to go with the approach here which is common.

Reply:
I thought of doing String comparisons. But, it then depends on the whims of the json serializing library. If it fails sometime in the future nobody will know why